OpenXmlPowerTools/DocumentAssembler.cs

```diff
@@ -605,10 +596,20 @@ static object ContentReplacementTransform(XNode node, XElement data, TemplateErr
                     }
                     catch (XPathException e)
                     {
-                        return CreateParaErrorMessage("XPathException: " + e.Message, templateError);
+                        return CreateContextErrorMessage(element, "XPathException: " + e.Message, templateError);
                     }
                     if (!repeatingData.Any())
-                        return CreateParaErrorMessage("Repeat: Select returned no data", templateError);
+                    {
+                        if (optional)
+                        {
+                            XElement para = element.Descendants(W.p).FirstOrDefault();
```

Comment:
@chrlutz Why are you creating an empty paragraph here?  It seems to me that if the repeat were optional, and if there were no data, the result would be that all of the content from the Repeat through the EndRepeat would simply not be included...